src/models/Factor.js

```diff
@@ -318,6 +321,21 @@ function (Okta, Q, factorUtil, Util, Errors, BaseLoginModel) {
 
     pushFactorHasAutoPush: function () {
       return this.settings.get('features.autoPush') && this.get('factorType') === 'push';
+    },
+
+    fixDataForCustomHotp: function(attributes) {
+      // Set vendorname from profile during enroll.
+      if (attributes.factorType === 'custom:hotp' && attributes.profiles) {
```

Comment:
@gowthamidommety-okta  @mauriciocastillosilva-okta  does this logic look good ? 

I will test it with actual backend integration on monday.